### PR TITLE
Added pause/resume for producers

### PR DESCRIPTION
**Pull request by accelerated**
Added similar helper functions to `consumer::pause` and `consumer::resume`.

### CODE REVIEW

**Comment by accelerated** [comment]
Not sure if it's worth caching the Topic object/partition list inside the producer so it can be reused when resume is called?

File: include/cppkafka/topic_partition_list.h [hunk 1]
```diff
@@ -37,6 +37,7 @@
 #include <set>
 #include <librdkafka/rdkafka.h>
 #include "macros.h"
+#include "metadata.h"
```
Comment (mfontanini) [inline include/cppkafka/topic_partition_list.h]: I would just use a forward declaration here, this pulls in a bunch of crap that I tried avoided pulling in other headers (e.g. `kafka_handle_base.h`)

**Comment by mfontanini** [comment]
Woah I didn't know pausing works for producers... Do you have a use case and do you know what happens when you produce into a paused topic/partition? Does it just get buffered for a while?

I feel like this should have an overload that takes `TopicPartitions`. I'm not a super fan of `get_metadata` but maybe I'm biased because I've faced at least 2 bugs in rdkafka related to this. Anyway, I imagine someone _may_ want to not produce into some particular topic/partition although I'm still not sure of what the use case would be for it.

File: include/cppkafka/topic_partition_list.h (hunk 1 above)
Comment (accelerated) [reply include/cppkafka/topic_partition_list.h]: ok will do

**Comment by accelerated** [comment]
Well apparently the rdkafka documentation for pause/resume does not involve a handle at all and it specifically refers to both production and consumption. For producers you pause when you get throttled. I'm currently implementing some throttling logic on top of cppkafka so i want to be able to pause on both ends. For the producer side, i assume that if you pause, the internal rdkafka queue won't get flushed at all, thus leaving the broker some time to breathe. During this time, all `sync/async_produce()` get routed to `add_message` instead which buffers locally in the `BufferedProducer`. When the throttle comes off, I `resume` and flush the buffer. 
No need to add an overload for `TopicPartitions`, you already have that function in KafkaHandleBase which takes a list. However I wanted to add an overload in the consumer to take a topic name. One use case is that i want to start some consumers in "paused" state and only start consuming when i'm ready. This is even before partitions get assigned so I would need a way to pause a `topic` rather than an "assigned partition list".

**Comment by accelerated** [comment]
I also assume that when pausing the consumer while continuing to poll it, only heartbeats go out and no more batch message requests.

**Comment by accelerated** [comment]
BTW posted a question for clarification in [librdkafka](https://github.com/edenhill/librdkafka/issues/1849)

**Comment by mfontanini** [comment]
Interesting. The author's answer regarding pausing unassigned partitions makes me wonder if this is a valid change. Like he says it _may_ work but they don't guarantee it does. Let's wait for him to answer your last question.

**Comment by accelerated** [comment]
Cheers!

**Comment by mfontanini** [comment]
Thanks for this one!

**Pull request merged**